src/rgw/driver/rados/rgw_rados.cc

```diff
@@ -3592,6 +3592,21 @@ int RGWRados::rewrite_obj(RGWBucketInfo& dest_bucket_info, const rgw_obj& obj, c
                        attrset, 0, real_time(), NULL, dpp, y);
 }
 
+int RGWRados::reindex_obj(RGWBucketInfo& dest_bucket_info, const rgw_obj& obj, const DoutPrefixProvider *dpp, optional_yield y)
+{
+  Bucket target(this, dest_bucket_info);
```

Comment:
probably a good place to return an error like `ENOTSUP` if `dest_bucket_info.versioned()`